src/k8s-extension/azext_k8s_extension/partner_extensions/CostExport.py

```diff
@@ -0,0 +1,263 @@
+    ]
+    cli = _cli()
+    # Not every subscription can create AmortizedCost (prefered) export
+    # If it fails, try to create Usage export
+    cli.invoke(args + ["--type", "AmortizedCost"])
+    if cli.result.exit_code == 1 and isinstance(cli.result.error,
+                                                HttpResponseError) and cli.result.error.status_code == 400:
+        logger.info("couldn't create AmortizedCost export, trying Usage")
+        cli = _cli()
+        cli.invoke(args + ["--type", "Usage"])
+        if cli.result.exit_code != 0:
+            raise cli.result.error
+    elif cli.result.exit_code == 0:
+        logger.info("created cost export with 'Usage' type")
+    else:
+        raise cli.result.error
+
+
+def _mc_resource_group(subscription: str, resource_group_name: str, cluster_name: str) -> str:
+    cli = _cli()
+    cli.invoke(["aks", "show",
+                "--subscription", subscription,
+                "--resource-group", resource_group_name,
+                "-n", cluster_name])
+    if cli.result.exit_code != 0:
+        raise Exception("Unable to get cluster resource group")
+    return cli.result.result['nodeResourceGroup']
+
+
+def _ensure_storage_container(storage_account_id: str, storage_container: str):
+    resource = parse_resource_id(storage_account_id)
+    try:
+        _invoke(["storage", "account", "show", "--ids", storage_account_id])
+    except SystemExit as e:
+        if e.code != 3:
+            raise e
+        create_storage_args = ["storage", "account", "create",
+                               "--resource-group", resource['resource_group'],
+                               "--name", resource['name'],
+                               "--subscription", resource['subscription'],
+                               "--allow-blob-public-access", "false"]
+        _invoke(create_storage_args)
+        logger.info("created storage account %s", storage_account_id)
+    cli = _invoke(["storage", "container", "exists", "--name", storage_container, "--account-name", resource["name"],
+                   "--auth-mode", "login"])
+    if cli.result.result['exists']:
+        return
+    _invoke(["storage", "container", "create", "--name", storage_container, "--account-name", resource["name"],
+             "--auth-mode", "login"])
+    logger.info("created new container %s for %s", storage_container, storage_account_id)
+
+
+class ServicePrincipal(TypedDict):
+    appId: str
+    password: str
+    tenant: str
+    displayName: str
+
+
+def _create_service_principal(sp_name: str) -> ServicePrincipal:
+    cli = _invoke(["ad", "sp", "create-for-rbac", "--display-name", sp_name, "--years", "2"])
+    logger.info("created service principal %s", sp_name)
+    return cli.result.result
+
+
+def _cli() -> AzCli:
+    return get_default_cli()
+
+
+def _invoke(args) -> AzCli:
+    cli = _cli()
+    try:
+        cli.invoke(args)
+    except SystemExit as e:
+        cmd = "az " + " ".join(args)
```

Comment:
```suggestion
        cmd = "az " + shlex.join(args)
```
Just in case there are spaces in any args.